src/core/api/track_choice_manager.ts

```diff
@@ -440,22 +467,36 @@ export default class TrackChoiceManager {
     }
 
     const videoAdaptations = period.getSupportedAdaptations("video");
-    const chosenVideoAdaptation = this._videoChoiceMemory.get(period);
 
-    if (chosenVideoAdaptation === null) {
-      // If the Period was previously without video, keep it that way
-      videoInfos.adaptation$.next(null);
-    } else if (chosenVideoAdaptation === undefined ||
-               !arrayIncludes(videoAdaptations, chosenVideoAdaptation)
-    ) {
-      const preferredVideoTracks = this._preferredVideoTracks;
-      const optimalAdaptation = findFirstOptimalVideoAdaptation(videoAdaptations,
-                                                                preferredVideoTracks);
-      this._videoChoiceMemory.set(period, optimalAdaptation);
-      videoInfos.adaptation$.next(optimalAdaptation);
+    const prevVideoAdaptation = this._videoChoiceMemory.get(period);
+    let newRootAdaptation : Adaptation | null;
```

Comment:
It is weird that typescript didn't scream here no?

Since, when we instantiate a new variable without assigning a new value it is set to `undefined` by default no?

Reply:
No, it's even done like that on purpose.

The idea is that this variable is for now unitialized and TypeScript will check that it is always used AFTER it has been initialized.

Thanks to this trick, you can do things like:
```ts
function getNum(str : "a"|"b"|"c") : number {
  let num : number; // not initialized there
  switch (str) {
    case "a":
      num = 1;
      break;
    case "b":
      num = 2;
      break;
    case "c":
      num = 3;
      break;
  }

  // TypeScript check that it is always initialized there
  return num + 1;
}
```

Yet things like that would throw an error:
```ts
function getNum(str : "a"|"b"|"c") : number {
  let num : number; // not initialized there
  switch (str) {
    case "a":
      num = 1;
      break;
  }

  // TypeScript fails here
  return num + 1;
}
```

Reply:
Oh okay understood!

Actually, typescript is smarter than I thought! 

Cool to learn new things :)